chronos/asyncloop.nim

```diff
@@ -227,14 +231,27 @@ func getAsyncTimestamp*(a: Duration): auto {.inline.} =
     result = cast[int32](res)
     result += min(1, cast[int32](mid))
 
+template removeTimers(loop) =
+  var timers = cast[seq[TimerCallback]](loop.timers)
```

Comment:
What was the goal here? This cast is quite naughty :)

Reply:
One problem is that removing items from the heap by looping modifies the internal `seq` and breaks everything. Tho as you point out bellow, this also breaks the binary heap.